File: Data Structures & Algorithms/anagram-groups/submission-4.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> res = new HashMap<>();
        for (String c: strs ){
            int[] occ= new int[26];
            for (int i=0; i<c.length();i++){
                occ[c.charAt(i)-'a']++;
            }
            String key = Arrays.toString(occ);
            res.putIfAbsent(key, new ArrayList<>());
            res.get(key).add(c);
        }
        return new ArrayList<>(res.values());
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-4.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        l=0
        while True : 

            for i in range(0,len(strs)):
                if l>=len(strs[i]) or strs[i][l]!=strs[0][l]:
                    return  strs[0][0:l]
            l+=1

        